src/dbos-executor.ts

```diff
@@ -777,6 +779,18 @@ export class DBOSExecutor implements DBOSExecutorContext {
           result = await retrievedHandle.getResult();
           wCtxt.span.setAttribute('cached', true);
           wCtxt.span.setStatus({ code: SpanStatusCode.OK });
+        } else if (err instanceof DBOSWorkflowCancelledError) {
+          internalStatus.error = err.message;
+          internalStatus.status = StatusString.CANCELLED;
+
+          if (!this.debugMode) {
+            await this.systemDatabase.setWorkflowStatus(workflowUUID, StatusString.CANCELLED, false);
+          }
+          wCtxt.span.setStatus({ code: SpanStatusCode.ERROR, message: err.message });
+          this.logger.info(`Cancelled workflow ${workflowUUID}`);
+
+          // debatable ? Another way to handle this is to throw the error and let the caller handle it.
```

Comment:
Yeah, I agree, we should throw the error and let the caller handle it. Separately, as you pointed out, we should make our starters more robust to errors--that's just good practice.